.github/actions/build_deb.sh

```diff
@@ -2,33 +2,67 @@
 
 echo "Building the SIL Kit packages"
 
-orig_tarball=${SILKIT_SOURCE_URL}/${SILKIT_VERSION}-${SILKIT_VERSION_SUFFIX}/libsilkit_${SILKIT_VERSION}.orig.tar.xz
-wget --no-proxy --no-check-certificate $orig_tarball
-ret_val=$?
-if [ "$ret_val" != '0' ] ; then
-    echo "SILKIT-PKG: Could not get orig tarball from $orig_tarball"
-    exit 64
+if [ -z $SILKIT_PKG_URL ] ; then
+    SILKIT_PKG_URL="."
 fi
+debian_path="${SILKIT_PKG_URL}/debian"
 
-mkdir -p libsilkit-${SILKIT_VERSION}
-tar -xvf libsilkit_${SILKIT_VERSION}.orig.tar.xz -C libsilkit-${SILKIT_VERSION}
-ret_val=$?
-if [ "$ret_val" != '0' ] ; then
-    echo "SILKIT-PKG: Could not untar orig tarball $orig_tarball"
-    exit 64
+silkit_pkg_found=false
+# Check if the provided path has a debian directory
+if [ -d ${debian_path} ] ; then
+    silkit_pkg_found=true
 fi
 
-git -c http.sslVerify=false clone ${SILKIT_PACKAGING_REPO} -b ${SILKIT_PACKAGING_REPO_BRANCH}
-ret_val=$?
-if [ "$ret_val" != '0' ] ; then
-    echo "SILKIT-PKG: Could not clone $SILKIT_PACKAGING_REPO : $SILKIT_PACKAGING_REPO_BRANCH"
-    exit 64
+if ! $silkit_pkg_found ; then
+    echo "SILKIT-PKG: Trying git sil-kit-pkg"
+    if git ls-remote -hq ${SILKIT_PKG_URL} ; then
+        git -c http.sslVerify=false clone --depth=1 ${SILKIT_PKG_URL} -b ${SILKIT_PKG_REF} sil-kit-pkg
+        ret_val=$?
+        if [ "$ret_val" != '0' ] ; then
+            echo "SILKIT-PKG: Could not clone $SILKIT_PKG_URL : $SILKIT_PKG_REF"
+            exit 64
+        fi
+        debian_path="./sil-kit-pkg/debian"
+    fi
 fi
 
-SILKIT_PACKAGE_REPO_NAME=$(echo ${SILKIT_PACKAGING_REPO} | awk -F/ '{print $NF}' | awk -F. '{print $1}')
-echo "PACKAGING REPO NAME ${SILKIT_PACKAGE_REPO_NAME}"
-mv ./${SILKIT_PACKAGE_REPO_NAME}/debian libsilkit-${SILKIT_VERSION}
 
+SILKIT_FULL_VERSION=`sed -En "s/libsilkit \(([0-9]+\.[0-9]+\.[0-9]+)-*([0-9]+).*/\1 \2/p" $debian_path/changelog`
+SILKIT_VERSION=`echo $SILKIT_FULL_VERSION | cut -d ' ' -f1`
```

Comment:
just code-golfing with posix sh here: there's another shorthand for stripping off parts of a string, `SILKIT_VERSION=${SILKIT_FULL_VERSION% *}` and `SILKIT_DEBIAN_REVISION=${SILKIT_FULL_VERSION:##* }`

What I really wanted to say is, do we need empty string checks here? or just put the whole script into `set -eu` mode to fail on unset variables.

Reply:
Will look into it. I think technically not all variables need to be set, e.g. SILKIT_PKG_URL is kinda optional. So `set -eu` would not work in that scenario. Unless we say `Github Action for sil-kit-pkg ref X can only build sil-kit ref X` and only accept `.` as SILKIT_PKG_URL

Reply:
The syntax for `SILKIT_DEBIAN_REVISION` did not work in Ubuntu dash. So keeping the old way for now :)